lib/Analysis/GateAnalysis.cc

```diff
@@ -171,24 +170,19 @@ void GateAnalysisImpl::processPhi(PHINode *PN, Instruction *insertionPt) {
   DenseMap<BasicBlock *, Value *> incomingBlockToValue =
       processIncomingValues(PN, insertionPt);
 
+  // Make sure CD blocks are sorted in reverse-topological order. We need this
+  // because we want to process them in order opposite to execution order.
+  auto bb_to_compare = [this](BasicBlock *first, BasicBlock *second) {
```

Comment:
Nit: use ThisCase for lambdas and give it a more descriptive name, e.g., `s/bb_to_compare/GreaterThanTopo`